render/wlr_renderer.c

```diff
@@ -276,6 +275,9 @@ struct wlr_renderer *wlr_renderer_autocreate(struct wlr_egl *egl,
 		config_attribs = all_config_attribs;
 	}
 
+	struct wlr_egl *egl = calloc(1, sizeof(*egl));
```

Comment:
This memory is leaked when the renderer is destroyed.

I think long-term we'll want to either:

* Allocate the `wlr_egl` as part of the `wlr_gles2_renderer`. This means `wlr_gles2_renderer_create` would need to take all arguments necessary for the call to `wlr_egl_init`.
* Make `wlr_gles2_renderer_create` take over the `wlr_egl` (kind of like you're doing in this PR), and replace `wlr_egl_init`/`wlr_egl_finish` with `wlr_egl_create`/`wlr_egl_destroy` so that the GLES2 renderer can cleanly free the `wlr_egl` (this would ensuring it's not part of a larger structure).

For this PR we can probably just `free(renderer->egl)` when destroying the GLES2 renderer, and properly document that `wlr_gles2_renderer_create` takes ownership of the `wlr_egl`?

Reply:
Huh, I was sure I added the `free` in the `wlr_renderer_destroy` function. Good catch.
I like the idea of `wlr_gles2_renderer` becoming the owner of `wlr_egl`, it makes more sense when we will add the pixman backend. 

I'll slap a nice comment and a TODO.

Reply:
OK. I slightly edited the PR to also add a note in the header defining `wlr_gles2_renderer_create`.